### PR TITLE
Migration from GCPMachineClass to MachineClass

### PR DESCRIPTION
**Pull request by prashanth26**
**What this PR does / why we need it**:
This PR allows migration of `GCPMachineClass` to `MachineClass` objects. It also updates any references to old objects.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

Check out the commits for ease of review

**Release note**:

```noteworthy user
Migration of `GCPMachineClass` to `MachineClass` CRs is implicity. Don't use old GCPMachineClass CRs post migration.
```

### CODE REVIEW

**Comment by prashanth26** [comment]
/invite @hardikdr @ggaurav10 @ashwani2k . Checkout commit-wise changes for ease of review.

**Comment by prashanth26** [comment]
cc: @AxiomSamarth . Similar changes for azure might be needed.

File: pkg/gcp/apis/provider_spec.go [hunk 1]
```diff
@@ -16,12 +16,17 @@ package api
 
 const (
 	// GCPServiceAccountJSON is a constant for a key name that is part of the GCP cloud credentials.
-	GCPServiceAccountJSON string = "serviceAccountJSON"
+	GCPServiceAccountJSON = "serviceAccountJSON"
+	// V1alpha1 is the API version
+	V1alpha1 = "mcm.gardener.cloud/v1alpha1"
```
Comment (hardikdr) [inline pkg/gcp/apis/provider_spec.go:21]: The term `V1alpha1` as s const sounds little confusing, we generally use it to refer to the API-package.

**Review by hardikdr — APPROVED** [review]
/lgtm just a minor nit, ignorable.

File: pkg/gcp/apis/provider_spec.go (hunk 1 above)
Comment (prashanth26) [reply pkg/gcp/apis/provider_spec.go]: Makes sense. I have changed this to - https://github.com/gardener/machine-controller-manager-provider-gcp/pull/5/files#diff-2d6dd4a52e0979ae4fd9992794d1861dR21. I hope it is less confusing now.

**Review by ggaurav10 — APPROVED** [review]

**Pull request merged**